src/yaml/builder.cr

```diff
@@ -244,7 +245,7 @@ module YAML
   end
 
   # Writes YAML into the given `IO`. A `YAML::Builder` is yielded to the block.
-  def self.build(io : IO)
+  def self.build(io : IO) : Nil
```

Comment:
No `io.flush` here?

Reply:
No, because `YAML::Builder` already does that.

Reply:
Why does `YAML::Builder` flush but `XML::Builder` doesn't? Shouldn't we make this behave identical?
`JSON::Builder#flush` delegates to `io.flush` which is another different behaviour.

Reply:
I noticed that too. I didn't want to start making changes all over the place. That's why I added `flush` when needed.

Also, maybe this PR is not needed...